qa/L0_java_sequence_batcher/SequenceTest.java

```diff
@@ -0,0 +1,640 @@
+            TRITONSERVER_InferenceRequest request, int flags, Pointer userp)
+        {
+          // We reuse the request so we don't delete it here.
+        }
+    }
+
+    static class InferResponseComplete extends TRITONSERVER_InferenceResponseCompleteFn_t {
+        @Override public void call (
+            TRITONSERVER_InferenceResponse response, int flags, Pointer userp)
+        {
+          if (response != null) {
+            // Send 'response' to the future.
+            futures.get(userp).complete(response);
+          }
+        }
+    }
+
+    static ConcurrentHashMap<Pointer, CompletableFuture<TRITONSERVER_InferenceResponse>> futures = new ConcurrentHashMap<>();
+    static ResponseAlloc responseAlloc = new ResponseAlloc();
+    static ResponseRelease responseRelease = new ResponseRelease();
+    static InferRequestComplete inferRequestComplete = new InferRequestComplete();
+    static InferResponseComplete inferResponseComplete = new InferResponseComplete();
+
+    static TRITONSERVER_Error
+    ParseModelMetadata(
+        JsonObject model_metadata,
+        boolean[] is_torch_model)
+    {
+      String seen_data_type = null;
+      for (JsonElement input_element : model_metadata.get("inputs").getAsJsonArray()) {
+        JsonObject input = input_element.getAsJsonObject();
+        if (!input.get("datatype").getAsString().equals("INT32")) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_UNSUPPORTED,
+              "sequence qa example only supports model with data type INT32");
+        }
+        if (seen_data_type == null) {
+          seen_data_type = input.get("datatype").getAsString();
+        } else if (!seen_data_type.equals(input.get("datatype").getAsString())) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_INVALID_ARG,
+              "the inputs and outputs of sequence model must have the data type");
+        }
+      }
+      for (JsonElement output_element : model_metadata.get("outputs").getAsJsonArray()) {
+        JsonObject output = output_element.getAsJsonObject();
+        if (!output.get("datatype").getAsString().equals("INT32")) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_UNSUPPORTED,
+              "sequence qa example only supports model with data type INT32");
+        } else if (!seen_data_type.equals(output.get("datatype").getAsString())) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_INVALID_ARG,
+              "the inputs and outputs of sequence' model must have the data type");
+        }
+      }
+
+      is_torch_model[0] =
+          model_metadata.get("platform").getAsString().equals("pytorch_libtorch");
+      return null;
+    }
+
+    // Custom function to set metadata required for
+    // sequence batcher
+    static void
+    SetSequenceMetadata(TRITONSERVER_InferenceRequest irequest, long correlation_id, boolean sequence_start, boolean sequence_end)
+    throws Exception
```

Comment:
do we need throws Exception here?

Reply:
Good catch (pun intended). Must have gotten mixed in during a previous commit... thanks, fixed!